Forge Networking Remastered Unity/Assets/Bearded Man Studios Inc/Scripts/Networking/Forge/Networking/FacepunchP2PServer.cs

```diff
@@ -0,0 +1,752 @@
+			Send(player, sendFrame, reliable);
+		}
+
+		/// <summary>
+		/// Sends binary message to all clients ignoring the specific one
+		/// </summary>
+		/// <param name="receivers">The clients / server to receive the message</param>
+		/// <param name="playerToIgnore"></param>
+		/// <param name="messageGroupId">The Binary.GroupId of the massage, use MessageGroupIds.START_OF_GENERIC_IDS + desired_id</param>
+		/// <param name="reliable">True if message must be delivered</param>
+		/// <param name="objectsToSend">Array of vars to be sent, read them with Binary.StreamData.GetBasicType<typeOfObject>()</param>
+		public virtual void Send(Receivers receivers = Receivers.Target, NetworkingPlayer playerToIgnore = null, int messageGroupId = MessageGroupIds.START_OF_GENERIC_IDS, bool reliable = false, params object[] objectsToSend)
+		{
+			BMSByte data = ObjectMapper.BMSByte(objectsToSend);
+			Binary sendFrame = new Binary(Time.Timestep, false, data, receivers, messageGroupId, false);
+			Send(sendFrame, reliable, playerToIgnore);
+		}
+
+		/// <summary>
+		/// Start the Forge Networking server on this Facepunch Steamworks client
+		/// </summary>
+		/// <param name="haveLobby">Boolean used internally to tell whether we have already created the lobby</param>
+		public void Host(bool haveLobby = false)
+		{
+			if (Disposed)
+				throw new ObjectDisposedException("FacepunchP2PServer", "This object has been disposed and can not be used to connect, please use a new FacepunchP2PServer");
+
+			try
+			{
+				if (!haveLobby)
+				{
+					CreateLobbyAsync();
+					return;
+				}
+
+				var selfSteamId = SteamClient.SteamId;
+				Client = new CachedFacepunchP2PClient(selfSteamId);
+				Me = new NetworkingPlayer(ServerPlayerCounter++, selfSteamId, true, this)
+				{
+					InstanceGuid = InstanceGuid.ToString()
+				};
+
+				// Do any generic initialization in result of the successful bind
+				OnBindSuccessful();
+
+				// Create the thread that will be listening for new data from connected clients and start its execution
+				Task.Queue(ReadClients);
+
+				// Create the thread that will check for player timeouts
+				Task.Queue(() =>
+				{
+					commonServerLogic.CheckClientTimeout((player) =>
+					{
+						Disconnect(player, true);
+						OnPlayerTimeout(player);
+						CleanupDisconnections();
+					});
+				});
+
+				// Let myself know I connected successfully
+				OnPlayerConnected(Me);
+				// Set myself as a connected client
+				Me.Connected = true;
+				OnServerCreated();
+				StartAcceptingConnections();
+			}
+			catch (Exception e)
+			{
+				Logging.BMSLog.LogException(e);
+				// Do any generic initialization in result of the binding failure
+				OnBindFailure();
+
+				throw new FailedBindingException("Failed to bind to SteamClient.SteamId, see inner exception", e);
+			}
+		}
+
+		/// <summary>
+		/// Creates a steam lobby
+		/// </summary>
+		/// <returns></returns>
+		private async void CreateLobbyAsync()
```

Comment:
Any reason to have this wrapper method to call the one below it? Can't `CreateLobby` be used directly?